src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts

```diff
@@ -0,0 +1,185 @@
+import { ChangeEvent, useState } from 'react';
+import { RadioChangeEvent } from 'antd';
+import { ITextareaWithDatesOnChange, OnChangeObjectKeys } from '../interfaces/interfaces';
+
+export const useOnDataChange = () => {
+  const categoryOptions = [
+    { id: 1, value: 'Legal' },
+    { id: 2, value: 'IT' },
+    { id: 3, value: 'Sales' },
+    { id: 4, value: 'Finance' },
+    { id: 5, value: 'Construction' },
+    { id: 6, value: 'Accounting' },
+    { id: 7, value: 'Design' },
+    { id: 8, value: 'Security' },
+    { id: 9, value: 'Healthcare' },
+    { id: 10, value: 'Marketing' },
+  ];
+
+  const [file, setFile] = useState<Blob>();
+  const [file64, setFile64] = useState<string | ArrayBuffer | null>();
+  const onChangePhotoHandler = async (e: ChangeEvent<HTMLInputElement>) => {
+    const reader = new FileReader();
+    reader.onloadend = () => {
+      setFile64(reader.result);
+    };
+    const newFile = e.target.files && e.target.files[0];
+    if (newFile) {
+      setFile(newFile);
+      reader.readAsDataURL(newFile);
+    }
+  };
+  const onPhotoDelete = () => {
+    setFile(undefined);
+    setFile64(null);
+  };
+
+  const [textAreaWithDatesState, setTextAreaWithDatesState] = useState({
+    education: [
+      {
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      },
+    ],
+    experience: [
+      {
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      },
+    ],
+  });
+  const onChangeTextareaWithDates = (args: ITextareaWithDatesOnChange) => {
+    setTextAreaWithDatesState(prevState => {
+      const newState = { ...prevState };
+      newState[args.key][args.index] = args.item;
+      console.log(newState);
+      return newState;
+    });
+  };
+  const addField = (key: OnChangeObjectKeys) => {
+    setTextAreaWithDatesState(prevState => {
+      const newState = { ...prevState };
+      newState[key].push({
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      });
+      return newState;
+    });
+  };
+
+  const [skillsOptions, setSkillsOptions] = useState([
```

Comment:
no, 1) move your moks to some mock constant after imports, 2d) data from backend will come without - true and false. I recomend set to state only selected items, and rest will come from backend. if you want go with way, anyway you will need map array that come from backend somewhere in useEffect to this new format, and then send to backend only selected items, which mean again transformation

Reply:
1) moved;
2) when I see the back api, I'll adapt my state